src/RemoteInboxNotifications/DataSourcePoller.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Handles polling and storage of specs
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+use Automattic\WooCommerce\Admin\Loader;
+
+/**
+ * Specs data source poller class.
+ * This handles polling specs from JSON endpoints, and
+ * stores the specs in to the database as an option.
+ */
+class DataSourcePoller {
+	const DATA_SOURCES = array(
+		'http://two.wordpress.test/rinds-specs.json',
+	);
+
+	/**
+	 * Polls the data sources for specs, persists those specs, and delete any
+	 * specs that no longer exist in the data sources.
+	 */
+	public static function poll_data_sources() {
+		if ( ! Loader::is_feature_enabled( 'remote-inbox-notifications' ) ) {
+			return;
+		}
+
+		$specs = array();
+
+		// Note that this merges the specs from the data sources based on the slug - last one wins.
+		foreach ( self::DATA_SOURCES as $url ) {
+			self::read_data_source( $url, $specs );
+		}
+
+		// Persist the specs as an option.
+		update_option( RemoteInboxNotificationsEngine::SPECS_OPTION_NAME, $specs );
+
+		// Run the engine.
+		RemoteInboxNotificationsEngine::run();
```

Comment:
It came to be in here because I expected to move it out into it's own cron task but that never happened. I've moved it into `Events.php` which is the only place the poller is run. Thanks :-)